### ISSUE TITLE
No access to shared folder

### ISSUE THREAD

**Issue by arnolddwarf** [ServerContainers/samba#112]
Hello.

I'm struggling to get the container to work, but no supply. Here I leave my compose file in case I am failing something

```dockerfile
  samba:
    build: .
    image: ghcr.io/servercontainers/samba
    container_name: samba
    restart: unless-stopped
    network_mode: host
    environment:
      MODEL: 'TimeCapsule'
      AVAHI_NAME: OPI
      SAMBA_CONF_LOG_LEVEL: 3
      GROUP_family: 0
      ACCOUNT_arnold: arnold123
      UID_arnold: 0
      GROUPS_arnold: family
      SAMBA_VOLUME_CONFIG_arnoldonly: "[arnold]; path=/shares/arnold; valid users = arnold; guest ok = no; read only = no; browseable = yes"
    volumes:
      - /mnt/opidata:/shares/arnold

``` 

Here the end of the log:

`samba  | [2023/12/01 06:18:52.723795,  0] ../../source3/nmbd/nmbd_become_lmb.c:398(become_local_master_stage2)
samba  |   *****
samba  |
samba  |   Samba name server ORANGEPI3B is now a local master browser for workgroup WORKGROUP on subnet 172.17.0.1
samba  |
samba  |   *****
samba  | [2023/12/01 06:18:52.724108,  3] ../../source3/nmbd/nmbd_namelistdb.c:257(add_name_to_subnet)
samba  |   add_name_to_subnet: Added netbios name WORKGROUP<1d> with first IP 192.168.100.44 ttl=0 nb_flags= 0 to subnet 192.168.100.44
samba  | [2023/12/01 06:18:52.724164,  3] ../../source3/nmbd/nmbd_become_lmb.c:354(become_local_master_stage2)
samba  |   become_local_master_stage2: registered as master browser for workgroup WORKGROUP on subnet 192.168.100.44
samba  | [2023/12/01 06:18:52.724186,  3] ../../source3/nmbd/nmbd_sendannounce.c:70(broadcast_announce_request)
samba  |   broadcast_announce_request: sending announce request for workgroup WORKGROUP to subnet 192.168.100.44
samba  | [2023/12/01 06:18:52.724384,  0] ../../source3/nmbd/nmbd_become_lmb.c:398(become_local_master_stage2)
samba  |   *****
samba  |
samba  |   Samba name server ORANGEPI3B is now a local master browser for workgroup WORKGROUP on subnet 192.168.100.44
samba  |
samba  |   *****
samba  | [2023/12/01 06:18:52.724916,  3] ../../source3/nmbd/nmbd_sendannounce.c:170(send_local_master_announcement)
samba  |   send_local_master_announcement: type 849a03 for name ORANGEPI3B on subnet 172.17.0.1 for workgroup WORKGROUP
samba  | [2023/12/01 06:18:52.725151,  3] ../../source3/nmbd/nmbd_sendannounce.c:189(send_workgroup_announcement)
samba  |   send_workgroup_announcement: on subnet 172.17.0.1 for workgroup WORKGROUP
samba  | [2023/12/01 06:18:52.725321,  3] ../../source3/nmbd/nmbd_sendannounce.c:170(send_local_master_announcement)
samba  |   send_local_master_announcement: type 849a03 for name ORANGEPI3B on subnet 192.168.100.44 for workgroup WORKGROUP
samba  | [2023/12/01 06:18:52.725476,  3] ../../source3/nmbd/nmbd_sendannounce.c:189(send_workgroup_announcement)
samba  |   send_workgroup_announcement: on subnet 192.168.100.44 for workgroup WORKGROUP
samba  | [2023/12/01 06:19:12.751809,  3] ../../source3/nmbd/nmbd_serverlistdb.c:414(write_browse_list)
samba  |   write_browse_list: Wrote browse list into file /var/cache/samba/browse.dat
samba  | [2023/12/01 06:21:02.873184,  3] ../../source3/nmbd/nmbd_sendannounce.c:170(send_local_master_announcement)
samba  |   send_local_master_announcement: type 849a03 for name ORANGEPI3B on subnet 172.17.0.1 for workgroup WORKGROUP

`

Please, could you tell me what I'm doing wrong, or maybe I'm writing the address wrong from my windows 10 (⧵⧵192.168.100.44⧵arnold)

**Comment by fgeck** [comment]
Hey @MarvAmBass 

re-opening this issue as I am desparetly looking for help.

I try the setup as described in the README.

Running the container on a debian VM in docker.

docker-compose.yaml:
```yaml
services:
  samba:
    image: ghcr.io/servercontainers/samba
    container_name: samba
    restart: always
    network_mode: host
    environment:
      MODEL: 'TimeCapsule'
      AVAHI_NAME: dockervm
      SAMBA_CONF_LOG_LEVEL: 3
      WSDD2_DISABLE: 1

      GROUP_fgeck: 1000
      ACCOUNT_fgeck: test
      GROUPS_fgeck: fgeck
      UID_fgeck: 1000
      SAMBA_VOLUME_CONFIG_public: "[data]; path=/mnt/data; valid users = fgeck; guest ok = yes; read only = no; 

    volumes:
      - /etc/avahi/services/:/external/avahi
      - /mnt/data:/mnt/data
```

I can access the VM by the static IP: 192.168.178.100
![image](https://github.com/ServerContainers/samba/assets/43751896/7a9d1c1e-369f-4c42-a41a-0ecb232ec416)

But when I use the credentials (fgeck / test) I get following error:
![image](https://github.com/ServerContainers/samba/assets/43751896/153cea42-def8-42be-87d0-48fd37dec01e)



Cheking the smbpasswd file in the container I can see that it is empty:
```bash
/ # ls -lah /var/lib/samba/private/
total 436K   
drwx------    1 root     root        4.0K Apr  4 14:50 .
drwxr-xr-x    1 root     root        4.0K Apr  4 14:50 ..
drwx------    2 root     root        4.0K Apr  4 14:50 msg.sock
-rw-------    1 root     root      420.0K Apr  4 14:50 secrets.tdb
-rw-------    1 root     root           0 Apr  4 14:50 smbpasswd
```

---

2nd approach was using the script `docker run -ti --rm --entrypoint create-hash.sh ghcr.io/servercontainers/samba`. Unfortunately error on windows stays the same and sambpasswd file stays empty:

```bash
 docker exec -ti samba cat /var/lib/samba/private/smbpasswd

```

**Comment by MarvAmBass** [comment]
hi I found a few mistakes in your config

```
services:
  samba:
    image: ghcr.io/servercontainers/samba
    container_name: samba
    restart: always
    network_mode: host
    environment:
      MODEL: 'TimeCapsule'
      AVAHI_NAME: dockervm
      SAMBA_CONF_LOG_LEVEL: 3
      WSDD2_DISABLE: 1

      ACCOUNT_fgeck: test
      UID_fgeck: 1000
      SAMBA_VOLUME_CONFIG_public: "[data]; path=/mnt/data; valid users = fgeck; guest ok = yes; read only = no; 

    volumes:
      - /etc/avahi/services/:/external/avahi
      - /mnt/data:/mnt/data
```

maybe this resolves the issue

**Comment by fgeck** [comment]
Thank you very much for that fast answer!

On a first try it did not solve the issue, but when I restarted the Windows machine I was able to access the samba share.
After that I changed the password to be used from my .env file - same problem "Permission Denied".
Restarted Windows again and I can access the samba share again :-)

So for anyone running into the same issue: 
if your conifg is 100% valid try restarting Windows. 
And in case someone has stored the wrong credentials, go to "Credentials Manager" -> "Windows Credentials" -> remove the ones from the samba share
